### ISSUE TITLE
Support for SpEL pointing to related entities in projection on first level.

### ISSUE THREAD

**Issue by venkat125** [spring-projects/spring-data-neo4j#2325]
Hi Team, i was trying out simple json response by mapping the nested DTO Projection values to the root Projection (node / object) by following this [SDN doc](https://docs.spring.io/spring-data/neo4j/docs/current/reference/html/#projections.interfaces.open)

```
public interface PersonProjection {
    String getName();
    
    @Value("#{target.currentLocation.city.name}")
    String getCityName();

    @Value("#{target.currentLocation.city.regionalState.name}")
    String getStateName();
}
```

**expected output:**

`{
    "name": "Dhoni",
    "cityName": "qwerqwer",
    "stateName": "zxvcxzcvc"
}`
But, it throws the following error stating currentLocation is null as below: (but it has value)

> WARN 13600 --- [nio-8080-exec-6] .w.s.m.s.DefaultHandlerExceptionResolver : Resolved [org.springframework.http.converter.HttpMessageNotWritableException: Could not write JSON: EL1007E: Property or field 'city' cannot be found on null; nested exception is com.fasterxml.jackson.databind.JsonMappingException: EL1007E: Property or field 'city' cannot be found on null (through reference chain: com.sun.proxy.$Proxy162["currentLocation"])]

For reference and brevity, i'm posting the [stackoverflow link](https://stackoverflow.com/questions/68353950/spring-data-neo4j-rest-projection-returns-null-when-using-spel-expression-on-nes).

where the 1st and 2nd case is working. But, not the 3rd case.
Is there anyway we can achieve this? Or correct me if i'm wrong.
Thanks Team.

**Comment by meistermeier** [comment]
Thanks for reporting this. The usage of only in the entity defined relationships' properties in SpEL was indeed a missing constellation we did not cover.
The other ones are working because the resolution already switched to look at the entities but it did not for the first level.
There is now a `6.1.3-GH-2325-SNAPSHOT` available. Would be really helpful if you could check if your project returns the right results with this.

To get the version, please add following to your pom
```
<dependency>
    <groupId>org.springframework.data</groupId>
    <artifactId>spring-data-neo4j</artifactId>
    <version>6.1.3-SNAPSHOT</version>
</dependency>
```
```
<repositories>
  <repository>
    <id>spring-milestones</id>
    <name>Spring Milestones</name>
    <url>https://repo.spring.io/milestone</url>
    <snapshots>
      <enabled>false</enabled>
    </snapshots>
  </repository>
  <repository>
    <id>spring-snapshots</id>
    <name>Spring Snapshots</name>
    <url>https://repo.spring.io/snapshot</url>
    <releases>
      <enabled>false</enabled>
    </releases>
  </repository>
</repositories>
```

**Comment by meistermeier** [comment]
Fixed with f23071b

**Comment by venkat125** [comment]
That works. Thanks @meistermeier
